src/libultra/voice/voicesetadconverter.c

```diff
@@ -1,3 +1,70 @@
-#include "global.h"
+/**
+ * File: voicesetadconverter.c
+ */
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/voicesetadconverter/__osVoiceSetADConverter.s")
+#include "ultra64/controller_voice.h"
+#include "io/controller.h"
+#include "functions.h"
+#include "variables.h"
+
+#define SWRITEFORMAT(ptr) ((__OSVoiceSWriteFormat*)(ptr))
+
+s32 __osVoiceSetADConverter(OSMesgQueue* mq, s32 channel, u8 data) {
+    s32 errorCode;
+    s32 i;
+    u8* ptr;
+    u8 status;
+    s32 retryCount = 2;
+
+    __osSiGetAccess();
+
+    do {
+        ptr = (u8*)&__osPfsPifRam.ramarray;
+
+        if ((__osContLastPoll != CONT_CMD_SWRITE_VOICE) || (channel != __osPfsLastChannel)) {
+            __osContLastPoll = CONT_CMD_SWRITE_VOICE;
+            __osPfsLastChannel = channel;
+
+            for (i = 0; i < channel; i++, *ptr++ = 0) {}
+
+            __osPfsPifRam.status = CONT_CMD_EXE;
+
+            ptr[0] = CONT_CMD_SWRITE_VOICE_TX;
+            ptr[1] = CONT_CMD_SWRITE_VOICE_RX;
+            ptr[2] = CONT_CMD_SWRITE_VOICE;
+            ptr[5] = 0;
```

Comment:
Oh I missed these. They should probably use the SWRITEFORMAT macro and set the appropriate struct fields.